compiler/src/main/kotlin/asmble/run/jvm/EnvModule.kt

```diff
@@ -0,0 +1,37 @@
+package asmble.run.jvm
+
+/**
+ * Module used for metering.
+ */
+open class EnvModule(val gasLimit: Long) {
+
+    private var overallSpentGas: Long = 0
+
+    /**
+     * [Wasm function]
+     * Adds spent gas to overall spent gas and checks limit exceeding.
+     */
+    fun gas(spentGas: Int) {
+        if(overallSpentGas + spentGas > gasLimit) {
+            // TODO : check for overflow, add exception
```

Comment:
We can return -1 instead of exception

Reply:
If you mean that a client Wasm code should check for this -1, it wouldn't work, because we can't trust it. And it seems that the exception here is the most straightforward way without changes in Asmble core.